### ISSUE TITLE
Error handling

### ISSUE THREAD

**Issue by flosse** [nickel-org/nickel-postgres#13]
In [line 49](https://github.com/nickel-org/nickel-postgres/blob/master/src/middleware.rs#L49) there are two `unwrap` calls:
```rust
self.extensions().get::<PostgresMiddleware>().unwrap().get().unwrap()
```
Could you explain what will happen if s.th. goes wrong?

**Comment by Ryman** [comment]
The first unwrap would be a programmer error (not adding the middleware before a handler - I plan to make that impossible but haven't gotten around to fixing it up).

The second unwrap would fail if there's a timeout getting a connection from the connection pool, that could definitely be improved by adding some kind of hook to handle it, or to adjust the api!

**Comment by flosse** [comment]
> that could definitely be improved by adding some kind of hook to handle it, or to adjust the api!

yes, that would be great! `Result` and `Option` is one of rusts strength...unfortunately `unwrap` exists too :(

**Comment by Ryman** [comment]
I think this can be considered fixed by #13.
